src/drivers/cmakeDriver.ts

```diff
@@ -979,27 +981,39 @@ export abstract class CMakeDriver implements vscode.Disposable {
     /**
      * Picks the best generator to use on the current system
      */
-    async findBestGenerator(preferredGenerators: CMakeGenerator[]): Promise<CMakeGenerator | null> {
+    async findBestGenerator(preferredGenerators: CMakeGenerator[], usingFallbackGenerators: boolean): Promise<CMakeGenerator | null> {
         log.debug(localize('trying.to.detect.generator', 'Trying to detect generator supported by system'));
         const platform = process.platform;
 
         for (const gen of preferredGenerators) {
             const gen_name = gen.name;
             const generator_present = await (async (): Promise<boolean> => {
-                if (gen_name === 'Ninja' || gen_name === 'Ninja Multi-Config') {
-                    return await this.testHaveCommand('ninja') || this.testHaveCommand('ninja-build');
-                }
-                if (gen_name === 'MinGW Makefiles') {
-                    return platform === 'win32' && this.testHaveCommand('mingw32-make');
-                }
-                if (gen_name === 'NMake Makefiles') {
-                    return platform === 'win32' && this.testHaveCommand('nmake', ['/?']);
-                }
-                if (gen_name === 'Unix Makefiles') {
-                    return this.testHaveCommand('make');
-                }
-                if (gen_name === 'MSYS Makefiles') {
-                    return platform === 'win32' && this.testHaveCommand('make');
+                // If we're not using the default generators, then the generator was manually specified
+                // in that case, assume it exists and don't check the PATH, only check if it's a supported generator.
+                // If we are using the default generators, then we should also check if it is on the PATH
+                if (!usingFallbackGenerators) {
+                    return gen_name === 'Ninja' ||
```

Comment:
If we're going back to, if the user has `cmake.generator` set, we use it without conditions, I don't think we should have these conditions here, we should simply return `true`.